packages/indexer/src/database/SpokePoolRepository.ts

```
@@ -0,0 +1,193 @@
+import winston from "winston";
+import * as across from "@across-protocol/sdk";
+import { getRelayHashFromEvent } from "@across-protocol/sdk/dist/cjs/utils/SpokeUtils";
+import {
+  DataSource,
+  ExecutedRelayerRefundRoot,
+  FilledV3Relay,
+  RelayedRootBundle,
+  RequestedV3SlowFill,
+  TokensBridged,
+  V3FundsDeposited,
+} from "@repo/indexer-database";
+
+export class SpokePoolRepository {
+  constructor(
+    private postgres: DataSource | undefined,
+    private logger: winston.Logger,
+  ) {}
+
+  private formatRelayData(
+    event:
+      | across.interfaces.DepositWithBlock
+      | across.interfaces.FillWithBlock
+      | across.interfaces.SlowFillRequestWithBlock,
+  ) {
+    return {
+      inputAmount: event.inputAmount.toString(),
+      outputAmount: event.outputAmount.toString(),
+      fillDeadline: new Date(event.fillDeadline * 1000),
+      exclusivityDeadline:
+        event.exclusivityDeadline === 0
+          ? undefined
+          : new Date(event.exclusivityDeadline * 1000),
+    };
+  }
+
+  public async formatAndSaveV3FundsDepositedEvents(
+    v3FundsDepositedEvents: across.interfaces.DepositWithBlock[],
+  ) {
+    const v3FundsDepositedRepository =
+      this.postgres?.getRepository(V3FundsDeposited);
+    const formattedEvents = v3FundsDepositedEvents.map((event) => {
+      return {
+        ...event,
+        relayHash: getRelayHashFromEvent(event),
+        ...this.formatRelayData(event),
+        quoteTimestamp: new Date(event.quoteTimestamp * 1000),
+      };
+    });
+    try {
+      await v3FundsDepositedRepository?.save(formattedEvents, { chunk: 2000 });
+      this.logger.info(
+        `Saved ${v3FundsDepositedEvents.length} V3FundsDeposited events`,
+      );
+    } catch (error) {
+      this.logger.error(
+        "There was an error while saving V3FundsDeposited events:",
+        error,
+      );
+    }
```

I don't think the error propagation should be stopped here. it should be up to the caller to decide if it wants to ignore the error or handle it. There's also the option to add an additional function parameter `throwError = true` if the caller doesn't want error propagation (will pass `false` as value)

---

I like the option of the `throwError` param. I implemented it here [5f0aaa0](https://github.com/across-protocol/indexer/pull/18/commits/5f0aaa0e0f6e2403d7a64b0740c6a9abab55c775).
I decided to set the default to false for now as every time you run the indexer now you will get duplicated events and the save functions will throw. I think we can change it to true once we work on improving the indexing loop.